app/src/main/java/com/example/myweatherapp/mvp/view/MyWeatherAppView.kt

```diff
@@ -3,13 +3,18 @@ package com.example.myweatherapp.mvp.view
 import android.app.Activity
 import android.widget.Toast
 import com.example.myweatherapp.R
+import com.example.myweatherapp.data.service.response.WeatherResponse
 import com.example.myweatherapp.databinding.ActivityMainBinding
 import com.example.myweatherapp.mvp.contract.MyWeatherAppContract
 
 class MyWeatherAppView(activity: Activity, private var binding: ActivityMainBinding) :
     ActivityView(activity), MyWeatherAppContract.MyWeatherAppView {
 
-    override fun showData() {
+    override fun showData(data:WeatherResponse) {
         Toast.makeText(context, R.string.conection_established, Toast.LENGTH_SHORT).show()
     }
+
+    override fun showError() {
+        TODO("Not yet implemented")
```

Comment:
if you keep it like with, will crash when you call this method

Reply:
+1

Reply:
![error without internet](https://user-images.githubusercontent.com/73836200/106645452-1f8a5e80-656b-11eb-8f8f-a2328a16c323.jpg)

Now there is a toast implemented, which informs the user that there is no conection

edit: i am having some problems with the emulators as you can se in the image